osbuild/pipeline.py

```diff
@@ -239,8 +239,26 @@ def run(self, tree, runner, build_tree, store, monitor, libdir, timeout=None):
         return BuildResult(self, r.returncode, r.output, api.metadata, api.error)
 
 
+class Runner:
+    def __init__(self, info, name: Optional[str] = None) -> None:
+        self.info = info
+
+        if not name:
+            name = os.path.basename(info.path)
+
+        self.name = name
```

Comment:
Stylistic and non-important: `self.name = name or os.path.basename(info.path)` (controversial).

Hoewver since this is comparing to a None value I would write out the full identity check (`if name is not None`).

Reply:
But I want `""` also be replaced by the filename. We have used `or` like this before, if nobody has a strong dislike against that, I will go with that.